source/tls_channel_handler.c

```diff
@@ -326,54 +299,87 @@ void aws_tls_ctx_options_set_minimum_tls_version(
     options->minimum_tls_version = minimum_tls_version;
 }
 
-/**
- * The following API is deprecated due to inconsistent override versus append system/library trust
- * behavior between Windows, Mac, and Unix based platforms
- */
 int aws_tls_ctx_options_override_default_trust_store_from_path(
     struct aws_tls_ctx_options *options,
     const char *ca_path,
     const char *ca_file) {
 
+    struct aws_string *ca_path_tmp = NULL;
+    struct aws_byte_buf ca_file_tmp;
+    AWS_ZERO_STRUCT(ca_file_tmp);
+
     if (ca_path) {
-        options->ca_path = aws_string_new_from_c_str(options->allocator, ca_path);
-        if (!options->ca_path) {
-            return AWS_OP_ERR;
+        if (options->ca_path) {
+            AWS_LOGF_ERROR(AWS_LS_IO_TLS, "static: cannot override trust store multiple times");
+            aws_raise_error(AWS_ERROR_INVALID_STATE);
+            goto error;
+        }
+
+        ca_path_tmp = aws_string_new_from_c_str(options->allocator, ca_path);
+        if (!ca_path_tmp) {
+            goto error;
         }
     }
 
     if (ca_file) {
-        if (aws_byte_buf_init_from_file(&options->ca_file, options->allocator, ca_file)) {
-            return AWS_OP_ERR;
+        if (options->ca_file.capacity) {
```

Comment:
why capacity vs. len?

Reply:
we discussed. I'm changing all `.len` and `.capacity` checks to use a helper function instead.

There were all kinds of bugs (leaks, unexpected behavior, etc) that could happen if someone passed in a 0-length file or cursor